matching-notification-api/src/main/java/org/phenotips/matchingnotification/storage/internal/DefaultMatchStorageManager.java

```diff
@@ -563,21 +592,41 @@ public boolean setUserContacted(PatientMatch match, boolean isUserContacted)
     }
 
     @Override
-    public boolean updateNotificationHistory(PatientMatch match, JSONObject notificationRecord)
+    public Collection<PatientMatch> updateNotificationHistory(PatientMatch match, JSONObject notificationRecord)
     {
         Session session = this.beginTransaction();
         boolean transactionCompleted = false;
 
+        Collection<PatientMatch> matchesToUpdate = new ArrayList<>();
+        matchesToUpdate.add(match);
+
+        if (!match.isLocal()) {
+            String referenceServerId = match.isIncoming() ? match.getReferenceServerId() : "";
```

Comment:
use `getStoredServerId()` instead of explicit check for incoming\outgoing:
`String referenceServerId = this.getStoredServerId(match.getReferenceServerId())`